contracts/fuelchain/FuelMessagePortal.sol

```diff
@@ -202,19 +188,28 @@ contract FuelMessagePortal is
                 abi.encodePacked(blockHeader.computeConsensusHeaderHash()),
                 blockInHistoryProof.proof,
                 blockInHistoryProof.key,
-                rootBlockHeader.height - 1
+                rootBlockHeader.height
             ),
             "Invalid block in history proof"
         );
 
-        //execute message
-        _executeMessageInHeader(message, blockHeader, messageInBlockProof);
-    }
+        //verify message in block
+        bytes32 messageId = CryptographyLib.hash(
+            abi.encodePacked(message.sender, message.recipient, message.nonce, message.amount, message.data)
+        );
+        require(
+            verifyBinaryTree(
+                blockHeader.outputMessagesRoot,
+                abi.encodePacked(messageId),
+                messageInBlockProof.proof,
+                messageInBlockProof.key,
+                blockHeader.outputMessagesCount
+            ),
+            "Invalid message in block proof"
+        );
 
-    /// @notice Gets the currently set timelock for all incoming messages (in milliseconds)
-    /// @return incoming message timelock
-    function incomingMessageTimelock() public view returns (uint64) {
-        return _incomingMessageTimelock;
+        //execute message
```

Comment:
I see what you're saying. The reason I like the comment is because I feel like it breaks things up into readable sections. With the color coding in most IDEs, it's a quick visual to break out this line from the rest of the code above. It could be a little more descriptive I guess, but I don't know what else to add, lol